stdlib/buffer.gr

```diff
@@ -96,674 +96,691 @@ let checkIsIndexInBounds = (i, len, buf) => {
+ * @param length: The number of bytes to append
+ * @param srcBuffer: The buffer to append
+ * @param dstBuffer: The buffer to mutate
  *
  * @since v0.4.0
  */
-export let setInt64 = (index, value, buffer) => {
-  checkIsIndexInBounds(index, _64BIT_LEN, buffer)
-  Bytes.setInt64(index, value, buffer.data)
+export let addBufferSlice = (start, length, srcBuffer, dstBuffer) => {
+  addBytesSlice(start, length, srcBuffer.data, dstBuffer)
 }
 
 /**
- * Appends a signed 64-bit integer to a buffer.
+ * @section Raw bytes: Functions for working with the raw bytes in a buffer.
+ */
+
+
+/**
+ * Gets a signed 8-bit integer starting at the given byte index.
  *
- * @param value: The value to set
- * @param buffer: The buffer to mutate
+ * @param index: The byte index to access
+ * @param buffer: The buffer to access
+ * @returns A 32-bit integer representing a signed 8-bit integer that starts at the given index
  *
  * @since v0.4.0
  */
-export let addInt64 = (value, buffer) => {
-  autogrow(_64BIT_LEN, buffer)
-  let index = buffer.len
-  buffer.len = buffer.len + _64BIT_LEN
-  setInt64(index, value, buffer)
+export let getInt8S = (index, buffer) => {
+  checkIsIndexInBounds(index, _8BIT_LEN, buffer)
+  Bytes.getInt8S(index, buffer.data)
 }
 
 /**
- * Gets a 64-bit float starting at the given byte index.
+ * Gets an unsigned 8-bit integer starting at the given byte index.
  *
  * @param index: The byte index to access
  * @param buffer: The buffer to access
- * @returns A 64-bit float that starts at the given index
+ * @returns A 32-bit integer representing an unsigned 8-bit integer that starts at the given index
  *
  * @since v0.4.0
  */
-export let getFloat64 = (index, buffer) => {
-  checkIsIndexInBounds(index, _64BIT_LEN, buffer)
-  Bytes.getFloat64(index, buffer.data)
+export let getInt8U = (index, buffer) => {
+  checkIsIndexInBounds(index, _8BIT_LEN, buffer)
+  Bytes.getInt8U(index, buffer.data)
 }
 
 /**
- * Sets a 64-bit float starting at the given byte index.
+ * Sets a signed 8-bit integer starting at the given byte index.
  *
  * @param index: The byte index to update
  * @param value: The value to set
  * @param buffer: The buffer to mutate
  *
  * @since v0.4.0
  */
-export let setFloat64 = (index, value, buffer) => {
-  checkIsIndexInBounds(index, _64BIT_LEN, buffer)
-  Bytes.setFloat64(index, value, buffer.data)
+export let setInt8 = (index, value, buffer) => {
+  checkIsIndexInBounds(index, _8BIT_LEN, buffer)
+  Bytes.setInt8(index, value, buffer.data)
 }
 
 /**
- * Appends a 64-bit float to a buffer.
+ * Appends a signed 8-bit integer to a buffer.
  *
  * @param value: The value to append
  * @param buffer: The buffer to mutate
  *
  * @since v0.4.0
  */
-export let addFloat64 = (value, buffer) => {
-  autogrow(_64BIT_LEN, buffer)
-  let index = buffer.len
-  buffer.len = buffer.len + _64BIT_LEN
-  setFloat64(index, value, buffer)
+export let addInt8 = (value, buffer) => {
+  addInt8help(value, buffer)
```

Comment:
These were updated to use the `help` functions that I refactored to the top of the file.